src/Client.ts

```diff
@@ -367,12 +367,15 @@ export class Client {
 
     /**
      * List files and directories in the current working directory.
+     *
+     * @param path The name of the remote file or directory. If undefined, will use current directory
      */
-    async list(): Promise<FileInfo[]> {
+    async list(path?: string): Promise<FileInfo[]> {
         await this.prepareTransfer(this)
         const writable = new StringWriter()
         const progressTracker = createNullObject() as ProgressTracker // Don't track progress of list transfers.
-        await download(this.ftp, progressTracker, writable, "LIST -a")
+        const command = path ? `LIST ${path}` : "LIST -a"
```

Comment:
Why does the option with `path` not include the argument `-a`?

Reply:
I had errors when specifying `-a` with the path. I might have previously tested with the `-a` after `path`, but I don't recall the exact problem. But I just tried again on two different servers and it's working well. 

I've updated the code to always use the `-a` and also added a call to `protectWhitespace`